src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/CLHSDB.java

```diff
@@ -149,13 +149,12 @@ private CLHSDB(String[] args) {
                 doUsage();
                 return;
             }
-            // If all numbers, it is a PID to attach to
-            // Else, it is a pathname to a .../bin/java for a core file.
             try {
+                // Attempt to attach as a PID
                 pid = Integer.parseInt(args[0]);
             } catch (NumberFormatException e) {
-                execPath = args[0];
-                coreFilename = "core";
+                // Attempt to connect to remote debug server
+                debugServerName = args[0];
```

Comment:
I assume this change to no longer support just specifying an execPath is not exposed to the user, unless they are invoking the CLHSDB class directly from the command line.

Reply:
Yes, but this code would be run from `SALauncher::runCLHSDB`, so we need to change.
https://github.com/YaSuenag/jdk/blob/JDK-8263342/src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/SALauncher.java#L282

Reply:
Yes, I understood that part. I just wanted to make sure, for example, that we weren't previously supporting something like `attach <exe_path>` in clhsdb, and that would be impacted by this, but I'm pretty sure clhsdb already prevents doing that. However, what if you used `jshdb <cmd> --exe` and don't specify a core file. Did that ever work, and if so will it still work?

Reply:
> Yes, I understood that part. I just wanted to make sure, for example, that we weren't previously supporting something like attach <exe_path> in clhsdb, and that would be impacted by this, but I'm pretty sure clhsdb already prevents doing that.

It would check at `SALauncher::buildAttachArgs` https://github.com/openjdk/jdk/blob/32cbd193d9655825d10e07be50ab3b7550ecb401/src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/SALauncher.java#L202-L213

> However, what if you used jshdb <cmd> --exe and don't specify a core file. Did that ever work, and if so will it still work?

Due to above code snippets, we can see the usage on the console in JDK 15 as following, and it is same behavior after this change.

```
$ jhsdb clhsdb --exe /java
You have to set --core.
    --pid <pid>             To attach to and operate on the given live process.
    --core <corefile>       To operate on the given core file.
    --exe <executable for corefile>

    The --core and --exe options must be set together to give the core
    file, and associated executable, to operate on. They can use
    absolute or relative paths.
    The --pid option can be set to operate on a live process.
    --core and --pid are mutually exclusive.

    Examples: jhsdb clhsdb --pid 1234
          or  jhsdb clhsdb --core ./core.1234 --exe ./myexe
```